hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -523,47 +423,27 @@ class Setting(object):
 )
 
 
-settings._define_setting(
-    "max_shrinks",
-    default=not_set,
-    description="""
-Passing ``max_shrinks=0`` disables the shrinking phase (see the ``phases``
-setting), but any other value has no effect and uses a general heuristic.
-""",
-    deprecation_message="""
-The max_shrinks setting has been disabled, as internal heuristics are more
-useful for this purpose than a user setting.
-""",
-    deprecated_since="2018-04-14",
-)
-
-
 def _validate_timeout(n):
+    assert n is not not_set, "Validation is skipped if the value is not_set"
     if n is unlimited:
-        return -1
-    else:
+        note_deprecation(
+            "The timeout setting has been removed.  Please delete it from your "
+            "settings, and optionally configure the `deadline` setting to replace it.",
+            since="RELEASEDAY",
+        )
         return n
+    raise InvalidArgument(
+        "The timeout setting has been removed.  "
+        "Please use the `deadline` setting instead."
+    )
 
 
 settings._define_setting(
     "timeout",
-    default=60,
-    description="""
-Once this many seconds have passed, falsify will terminate even
-if it has not found many examples. This is a soft rather than a hard
-limit - Hypothesis won't e.g. interrupt execution of the called
-function to stop it. If this value is <= 0 then no timeout will be
-applied.
-""",
-    hide_repr=False,  # Still affects behaviour at runtime
-    deprecation_message="""
-The timeout setting is deprecated and will be removed in a future version of
-Hypothesis. To get the future behaviour set ``timeout=hypothesis.unlimited``
-instead (which will remain valid for a further deprecation period after this
-setting has gone away).
-""",
-    deprecated_since="2017-11-02",
-    future_default=unlimited,
+    default=not_set,
+    description="A deprecated setting.  Use `deadline` instead.",
```

Comment:
I don't think this is a good deprecation message. `timeout` and `deadline` are not at all equivalent - they do very different things. I'd be inclined to have a message more like "timeout has been deprecated and no longer does anything. This setting can safely be removed with no effect."